test/misc/unit/libraries/PricingLib.t.sol

```diff
@@ -139,54 +437,131 @@ contract PurePricingLibTest is Test {
         D18 pricePoolPerAsset = d18(uint128(bound(pricePoolPerAsset_, MIN_PRICE, MAX_PRICE_POOL_PER_ASSET)));
         D18 pricePoolPerShare = d18(uint128(bound(pricePoolPerShare_, MIN_PRICE, MAX_PRICE_POOL_PER_SHARE)));
 
-        uint256 expectedAssetAmount = pricePoolPerAsset.reciprocalMulUint256(
-            pricePoolPerShare.mulUint256(shareAmount, ROUNDING_DOWN).mulDiv(10 ** assetDecimals, 10 ** POOL_DECIMALS),
-            ROUNDING_DOWN
+        uint256 underestimate = pricePoolPerAsset.reciprocalMulUint256(
+            pricePoolPerShare.mulUint256(
+                uint256(shareAmount).mulDiv(10 ** assetDecimals, 10 ** SHARE_DECIMALS), MathLib.Rounding.Down
+            ),
+            MathLib.Rounding.Down
+        );
+        uint256 expectedDown = MathLib.mulDiv(
+            shareAmount * 10 ** assetDecimals,
+            pricePoolPerShare.inner(),
+            10 ** SHARE_DECIMALS * pricePoolPerAsset.inner(),
+            MathLib.Rounding.Down
+        );
+        uint256 expectedUp = MathLib.mulDiv(
+            shareAmount * 10 ** assetDecimals,
+            pricePoolPerShare.inner(),
+            10 ** SHARE_DECIMALS * pricePoolPerAsset.inner(),
+            MathLib.Rounding.Up
         );
 
         uint256 result = PricingLib.shareToAssetAmount(
-            shareAmount, POOL_DECIMALS, assetDecimals, pricePoolPerAsset, pricePoolPerShare, ROUNDING_DOWN
+            shareAmount, POOL_DECIMALS, assetDecimals, pricePoolPerAsset, pricePoolPerShare, MathLib.Rounding.Down
         );
 
-        assertEq(result, expectedAssetAmount, "shareToAssetAmount failed");
+        assertGe(result, underestimate, "shareToAssetAmount should be at least as large as underestimate");
```

Comment:
@lemunozm One example that proves `priceA.reciprocal(priceB.mul(amount))` is less accurate than `Math.mulDiv(x, y, z)`.